tldw_Server_API/app/core/AuthNZ/federation/provisioning_service.py

```
@@ -0,0 +1,683 @@
+                or revoke_roles
+            ),
+            "grant_org_ids": grant_org_ids,
+            "grant_team_ids": grant_team_ids,
+            "grant_roles": grant_roles,
+            "revoke_org_ids": revoke_org_ids,
+            "revoke_team_ids": revoke_team_ids,
+            "revoke_roles": revoke_roles,
+            "warnings": warnings,
+        }
+
+    async def apply_mapped_grants(
```

Ruff (0.15.6)

[warning] Too many branches (30 > 12)

(PLR0912)